### PR TITLE
feat(ci/cd): main build

### PR DESCRIPTION
**Pull request by paologallinaharbur**
Using bash if-else since github action is not supporting if-else statements.

### CODE REVIEW

File: .github/workflows/nightlyAndMainImage.yml [hunk 1]
```diff
@@ -50,11 +47,17 @@ jobs:
...
     env:
       DOCKER_IMAGE_NAME: newrelic/nri-kubernetes
-      DOCKER_IMAGE_TAG: nightly
       DOCKER_PLATFORMS: "linux/amd64,linux/arm64,linux/arm" # Must be consistent with the matrix from the job above
     steps:
       - uses: actions/checkout@v2
-
+      - name: Setting Docker image tag
+        run: |
+          if [[ "${{ github.ref_name }}" == "main" ]]
```
Comment (roobre) [inline .github/workflows/nightlyAndMainImage.yml:55]: What is `ref_name` and how is it used to distinguish cron/merge?

File: .github/workflows/nightlyAndMainImage.yml [hunk 2]
```diff
@@ -50,11 +47,17 @@ jobs:
...
+          then
+          echo "DOCKER_IMAGE_TAG=main" >> $GITHUB_ENV
+          else
+          echo "DOCKER_IMAGE_TAG=nightly" >> $GITHUB_ENV
+          fi
```
Comment (roobre) [inline .github/workflows/nightlyAndMainImage.yml:60]: I would use a more compact syntax and indentation

```suggestion
          if [[ "${{ github.ref_name }}" == "main" ]]; then
            echo "DOCKER_IMAGE_TAG=main" >> $GITHUB_ENV
          else
            echo "DOCKER_IMAGE_TAG=nightly" >> $GITHUB_ENV
          fi
```

File: .github/workflows/nightlyAndMainImage.yml (hunk 1 above)
Comment (paologallinaharbur) [reply .github/workflows/nightlyAndMainImage.yml:55]: somehow I assumed without even thinking about that, that the ref_name was going to be different. The assomption should be completely wrong. 🙇 

I will check that

Comment (roobre) [reply .github/workflows/nightlyAndMainImage.yml:55]: In other workflows I've used `github.event`, as will have an object with the name of the event type.

For example, `${{ if github.event.schedule }}` should evaluate to true if the event runs on schedule:

https://github.com/newrelic/nri-kubernetes/blob/a0ed2ac4810af825bd8b9f69f3fdd1692ad7c697/.github/workflows/security.yaml#L23

Comment (roobre) [reply .github/workflows/nightlyAndMainImage.yml:55]: I'm not sure how that could be put in a bash `if`... but I guess we can always have two steps setting the variable, and `skip` based on that :hack:

Comment (paologallinaharbur) [reply .github/workflows/nightlyAndMainImage.yml]: I added a default and a set that can be skipped

Comment (paologallinaharbur) [reply .github/workflows/nightlyAndMainImage.yml]: @roobre It feels like a bit an hack, let me know what you think

File: .github/workflows/nightlyAndMainImage.yml [hunk 3]
```diff
@@ -1,14 +1,11 @@
-name: Nightly build
+name: Nightly and Main build
```
Comment (roobre) [inline .github/workflows/nightlyAndMainImage.yml:1]:
```suggestion
name: Nightly and main build
```

**Review by roobre — APPROVED** [review]

**Pull request merged**